src/Tizen.BatteryMonitor/Tizen.BatteryMonitor/BatteryMonitor.cs

```diff
@@ -0,0 +1,206 @@
+﻿/*
+ * Copyright (c) 2020 Samsung Electronics Co., Ltd All Rights Reserved
+ *
+ * Licensed under the Apache License, Version 2.0 (the License);
+ * you may not use this file except in compliance with the License.
+ * You may obtain a copy of the License at
+ *
+ * http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an AS IS BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ */
+
+using System;
+using System.Collections.Generic;
+using System.Runtime.InteropServices;
+
+namespace Tizen.BatteryMonitor
+{
+    static internal class Globals
+    {
+        internal const string LogTag = "Tizen.BatteryMonitor";
+    }
+
+    /// <summary>
+    /// Provides information related to the power consumption by applications or by hardware resources on a battery-powered device for a given duration of time.
+    /// </summary>
+    /// <since_tizen> 6 </since_tizen>
+    public sealed class BatteryMonitor : IDisposable
+    {
+        private static volatile BatteryMonitor _instance = null;
+        private static readonly object _syncLock = new object();
+        private bool _disposed = false;
+
+        /// <summary>
+        /// The constructor of the BatteryMonitor class.
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        /// <feature>http://tizen.org/feature/battery</feature>
+        /// <privilege>http://tizen.org/privilege/systemmonitor</privilege>
+        private BatteryMonitor()
+        {
+        }
+
+        /// <summary>
+        /// Returns an instance of BatteryMonitor class
+        /// </summary>
+        /// <since_tizen> 6 </since_tizen>
+        public static BatteryMonitor Instance
+        {
+            get
+            {
+                if (_instance != null) return _instance;
+                lock (_syncLock)
+                {
+                    if (_instance == null) {
```

Comment:
Recommend to use `Lazy<>`

Reply:
private static volatile Lazy<BatteryMonitor> _instance = new Lazy<BatteryMonitor>(() => new BatteryMonitor());

Can i make the _instance variable volatile instead of readonly ?
because need to make it null in Dispose() method.

Reply:
If you want to manage a lifetime of object as possible as shortly, do not provide object as singleton,
if not,  use Lazy with permanently lifetime. 
`shorten lifetime` and `singleton` is not pair